### DISCUSSION TITLE
Embeddings model: bge-base-en-v1.5

### DISCUSSION THREAD

**Discussion by francis2tm** [ggml-org/llama.cpp#3667] [Q&A] [answered]
Hello,
I was wondering if it's possible to run bge-base-en-v1.5 model with llama.cpp, and if yes, could anyone give me a breakdown on how to do it?

Thanks in advance!

**Comment by ianscrivener** [comment]
**bge-base-en-v1.5** is not on the list of supported models. 

Curious, what about this model makes it better than some of the many other models that [llama.cpp already supports](https://github.com/ggerganov/llama.cpp#description)?

**Reply by francis2tm** [reply]
Indeed @ianscrivener, but this is an **embeddings model**. And a quite good at that: https://huggingface.co/spaces/mteb/leaderboard, currently the 2nd best embeddings model.
I wouldn't mind doing the support myself and later do a PR. But I would love a little guidance beforehand ;)

**Reply by ianscrivener** [reply]
Embeddings model. Right. I knew that last week but seem to have experienced a grey matter brown out 😳

May be you can tweak one of the [examples](https://github.com/ggerganov/llama.cpp/tree/master/examples) as a POC?

Hopefully one of the C/C++ devs can chime in... this is above my pay grade.

**Comment by monatis** [comment]
Supporting embedding models is on the way, starting from BERT and its variants. And it will be accompanied by a RAG example directly in C/C++.

**Reply by francis2tm** [reply]
Awesome! Do you have a timeline for the RAG example?

**Comment by monatis** [comment, marked as answer]
Estimating one week or so.

**Reply by zerodrift** [reply]
Hi there. Any updates on this?  sorry to bother you  ... :)

**Reply by Johnhersh** [reply]
Kinda curious what happened with this?

**Reply by willie** [reply]
Oh man, what happened?

**Comment by xyzhang626** [comment]
hey @francis2tm @zerodrift and other guys, if you are still interested, I've implemented BGE series model with ggml. Check [this embeddings.cpp](https://github.com/xyzhang626/embeddings.cpp.git).

**Reply by golangpkg** [reply]
can you merge embeddings.cpp into llama.cpp ?

**Reply by xyzhang626** [reply]
hey @golangpkg thanks for your interest. I believe there are some awesome guys have done this. Check [this](https://github.com/jordankanter/llama.cpp/commit/eea4ecc02089beaee8bd5e650ae7e946585455b6).

**Reply by phymbert** [reply]
The server has an embeddings endpoint openAI compatible. What do you need more ?

**Comment by wangzhaode** [comment]
hey @francis2tm
I've implemented bge-large-zh in project  [mnn-llm](https://github.com/wangzhaode/mnn-llm/blob/master/demo/embedding_demo.cpp)

bge-large-zh 8bit quant model speed can reach 749.12 tok/s on  Apple M1 Soc with 4-thread.
